solana-programs/claimable-tokens/program/tests/tests.rs

```diff
@@ -623,18 +686,291 @@ async fn transfer_with_amount_instruction_secp_index_exploit() {
 
     assert!(tx_result.is_err());
 
-    let bank_token_account_data = get_account(&mut program_context, &user_bank_account).await;
+    let bank_token_account_data = get_account(&mut program_context, &user_bank_account)
+        .await
+        .unwrap();
     let bank_token_account =
         spl_token::state::Account::unpack(&bank_token_account_data.data.as_slice()).unwrap();
     // check that program failed to send by validating data
     assert_eq!(bank_token_account.amount, tokens_amount);
-    let user_token_account_data =
-        get_account(&mut program_context, &user_token_account.pubkey()).await;
+    let user_token_account_data = get_account(&mut program_context, &user_token_account.pubkey())
+        .await
+        .unwrap();
     let user_token_account =
         spl_token::state::Account::unpack(&user_token_account_data.data.as_slice()).unwrap();
     assert_eq!(user_token_account.amount, 0);
 }
 
+// Test to verify signed amount requirement
+// Ensures that a different amount from the signed data cannot be invoked
+#[tokio::test]
+async fn transfer_amount_mismatch() {
+    let mut program_context = program_test().start_with_context().await;
+    let rent = program_context.banks_client.get_rent().await.unwrap();
+    let (
+        _rng,
+        _key,
+        priv_key,
+        _secp_pubkey,
+        mint_account,
+        mint_authority,
+        user_token_account,
+        eth_address,
+    ) = init_test_variables();
+
+    let mint_pubkey = mint_account.pubkey();
+    let (base_acc, user_bank_account, _) = prepare_transfer(
+        &mut program_context,
+        mint_account,
+        rent,
+        mint_authority,
+        eth_address,
+        &user_token_account,
+    )
+    .await;
+    // Transfer a single token for reuse
+    let transfer_amount = 1;
+    let nonce_acct_seed = [NONCE_ACCOUNT_PREFIX.as_ref(), eth_address.as_ref()].concat();
+    let (_, nonce_account, _) = find_nonce_address(&id(), &mint_pubkey, &nonce_acct_seed);
+
+    let current_user_nonce = get_user_account_nonce(&mut program_context, &nonce_account).await;
+    let transfer_instr_data = TransferInstructionData {
+        target_pubkey: user_token_account.pubkey(),
+        amount: transfer_amount + 1,
```

Comment:
this seems like a bit of a subtle way to test this, can we do

let stated_transfer_amount = 1
let attempted_transfer_amount = 1000

and just use vars?